mice-and-cats-java/src/main/java/foop/world/World.java

```diff
@@ -306,36 +270,52 @@ public void render(Graphics2D g, int w, int h, String playerName, boolean superV
                     g.drawString(name, x, y);
                 }
             }
+
+            // if the player is underground and was on the surface once draw the last cat position
+            if (playerEntity.isUnderground() && countOnSameSubway > countInMySubway) {
+                // update cat last seen position
+                if (isCat) {
+                    catLastSeen.put(entity.getId(), entity.getPosition());
+                }
+            }
```

Comment:
I think doing this in the render method is a bit weird, because if a mouse enters and immediately leaves again (or if one enters and one leaves between frames) the update could potentially be missed. World.entityUpdate might have been a better place, but I guess this is fine as well.